CMakeLists.txt

```diff
@@ -36,8 +41,16 @@ if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build, options are: Debug RelWithDebInfo." FORCE)
 endif()
 set(VERSION_TAG "" CACHE STRING "Build date for nightly versioning")
-set(EXTRA_DATA_DIR "" CACHE STRING "Directory for externalized data")
```

Comment:
The old default in cmake may be "", but in code we then assume it's in the bin path — if the path is unset. This now changes ...

Reply:
Yes, but only for linux and it's documented in the README.

Reply:
Also for BSDs. But the main question is basically whether this still works for running the game from the build dir without setting EXTRA_DATA_DIR. The previous default was to look in the cwd and now it's a different path.

Reply:
> But the main question is basically whether this still works for running the game from the build dir without setting EXTRA_DATA_DIR. The previous default was to look in the cwd and now it's a different path.

Yes. It can still be set to empty so it looks in cwd. The default changed on linux and is documented in COMPILATION.md. The documentation could be expanded to explain that an empty variable means that cwd is checked (which it currently doesn't explain).

Reply:
You miss the point. If I understand the change correctly, there is now an extra thing to remember when compiling — most of the developers are on linux and there is no need to pass any cmake options now. Changing that just makes contributing harder. It's the packagers that are the special case here.

Reply:
Well, the way it is now is an utter mess. If you want to make contributing easier, you better introduce a developer build option or utilize [CMAKE_BUILD_TYPE](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html) properly instead of having people micro-manage specific variables.

Reply:
There's no requirement for micromanagement at all.

Reply:
Ok, we obviously won't agree on anything. The build system is in a state where it cannot be used by linux users or packagers properly, since it doesn't adhere to any standards. It hardcodes paths on cmale level and has broken path handling on code level. Good luck!